crates/matrix-sdk-ui/src/timeline/event_item/mod.rs

```diff
@@ -158,9 +158,21 @@ impl EventTimelineItem {
         let event_id = event.event_id().to_owned();
         let is_own = client.user_id().map(|uid| uid == sender).unwrap_or(false);
 
+        // Get the room's power levels for calculating the latest event
+        let power_levels = if let Some(room) = client.get_room(room_id) {
+            room.power_levels().await
+        } else {
+            None
+        };
+        let room_power_levels_info = match (client.user_id(), power_levels.as_ref()) {
+            (Some(user_id), Some(power_levels)) => Some((user_id, power_levels)),
+            _ => None,
+        };
```

Comment:
GOOD NEWS EVERYONE! There's [`Option::zip()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.zip) that does exactly the same thing :)

```suggestion
        let room_power_levels_info = client.user_id().zip(power_levels.as_ref()));
```